Description: search multiarch paths for libraries
 hack to get numpys distutils multiarch aware
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/python-numpy/+bug/818867
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=640940
Author: Julian Taylor <jtaylor@ubuntu.com>
Forwarded: no (it's a Debian specific patch, for now)

--- a/numpy/distutils/system_info.py
+++ b/numpy/distutils/system_info.py
@@ -206,6 +206,16 @@ else:
     default_x11_include_dirs = ['/usr/X11R6/include', '/usr/X11/include',
                                 '/usr/include']
 
+    try:
+       import subprocess as sp
+       p = sp.Popen(["gcc", "-print-multiarch"], stdout=sp.PIPE)
+       triplet = str(p.communicate()[0].decode().strip())
+       if p.returncode == 0:
+           default_x11_lib_dirs += [os.path.join("/usr/lib/", triplet)]
+           default_lib_dirs += [os.path.join("/usr/lib/", triplet)]
+    except:
+       warnings.warn('failed to detect multiarch paths, please install gcc')
+
     if os.path.exists('/usr/lib/X11'):
         globbed_x11_dir = glob('/usr/lib/*/libX11.so')
         if globbed_x11_dir:
